### PR TITLE
Feature/351 revisions api

### LINKED ISSUE #351: Add zome API endpoints for retrieving previous record revisions by `ActionHash`
API methods to retrieve specific (stale) revisions of records must be implemented for all relevant record types (basically exposing `read_record_entry_by_header` to the frontend). These should be linked to `revision(revisionId: ID!)` query edges for all record types (this is part of `VfModule.History`).

### PR DESCRIPTION
**Pull request by pospi**
closes #351

### CODE REVIEW

**Comment by pospi** [comment]
Weird that this is timing out on some of the Ubuntu checks but not the MacOS one now. @Connoropolous any thoughts about that? Or do you think we should consider it working?

**Comment by Connoropolous** [comment]
"Weird that this is timing out on some of the Ubuntu checks but not the MacOS one now. @Connoropolous any thoughts about that? Or do you think we should consider it working?"

I am not concerned about timeouts, so if any one of the 4 runs passes then I consider it a pass.

File: zomes/rea_agent/zome/src/lib.rs [hunk 1]
```diff
@@ -47,6 +47,11 @@ fn get_agent(ByAddress { address }: ByAddress<AgentAddress>) -> ExternResult<Res
     Ok(handle_get_agent(address)?)
 }
 
+#[hdk_extern]
+fn get_revision(ByRevision { revision_id }: ByRevision) -> ExternResult<ResponseData> {
+    Ok(handle_get_revision(revision_id)?)
```
Comment (Connoropolous) [inline zomes/rea_agent/zome/src/lib.rs:52]: I've been wondering if it makes any difference to write it this way, versus just: 
handle_get_revision(revision_id)

**Review by Connoropolous — APPROVED** [review]
All looks good. It's mostly just the same changes over and over again for each record type. I read over everything though 👍

File: zomes/rea_agent/zome/src/lib.rs (hunk 1 above)
Comment (pospi) [reply zomes/rea_agent/zome/src/lib.rs:52]: I could never get `Error`/`Result` conversion traits to work without the unwrap/rewrap. If you are able to, I'm interested in making it more succinct :wink:

**Pull request merged**